src/main/java/com/dezorganizacja/bridge/view/graphical/FXApplication.java

```diff
@@ -16,14 +23,60 @@ public static void setMainPresenter(MainPresenter mainPresenter) {
 
     @Override
     public void start(Stage stage) throws Exception {
-        FXMLLoader loader = new FXMLLoader(getClass().getResource("/fxml/main.fxml"));
-        Parent root = loader.load();
+        FXApplication.stage = stage;
+        FXApplication.mainPresenter.addObserver(this);
+        loadMainScene();
+    }
+
+    private void loadMainScene() {
+        String programState = mainPresenter.getState();
+        String fxml = stateToFXML(programState);
+
+        if (FXApplication.fxml != null && FXApplication.fxml.equals(fxml)) {
+            return;
+        }
+
+        FXMLLoader loader = new FXMLLoader(getClass().getResource(fxml));
+        Parent root;
+        try {
+            root = loader.load();
+        } catch (IOException e) {
+            System.err.println("error while loading " + fxml + " resource");
+            e.printStackTrace();
+            return;
+        }
+
         stage.setTitle("Bridge");
         stage.setScene(new Scene(root, 300, 275));
 
-        MainController controller = loader.getController();
-        controller.init(mainPresenter);
+        Controller controller = loader.getController();
+        if (controller != null) {
+            controller.init(mainPresenter);
+        }
 
         stage.show();
     }
+
+    public void changeMainScene() {
+        // TODO thread safety
```

Comment:
changeMainScene kolejkuje tylko zmianę sceny głównej, muszę doczytać jeszcze, czy to wystarczy, czy trzeba dorobić do tego zapewnienie, że nie zmieni się scena na już nieaktualną